src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -284,8 +284,8 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			this._updateSetup(folderSetup);
 			return this._updateWorkspaceTasks(TaskRunSource.FolderOpen);
 		}));
-		this._register(this._configurationService.onDidChangeConfiguration(() => {
-			if (!this._taskSystem && !this._workspaceTasksPromise) {
+		this._register(this._configurationService.onDidChangeConfiguration((e) => {
+			if (!e.affectedKeys.has('tasks.tasks') || (!this._taskSystem && !this._workspaceTasksPromise)) {
```

Comment:
I think you might want just `tasks` here. If I understand correctly, `tasks.tasks` will just be the `tasks` key in the `tasks.json` file. This is too narrow since you can have other keys at the same level as the `tasks` key that would cause the tasks to change.

Reply:
Thanks, this seemed wrong to me, so wanted to check

Reply:
Just `tasks` doesn't seem to work here when I change the `tasks.json` file. I wonder if we cannot use `affectsKeys` for this case 🤔 . The key that comes in for that is `tasks.tasks`